lldb/test/API/functionalities/reverse-execution/main.c

```diff
@@ -0,0 +1,12 @@
+static void start_recording() {}
+
+static void trigger_breakpoint() {}
+
+static void stop_recording() {}
```

Comment:
It would be nice to add some sample code in each of these functions to test of lldb behaves with things like variable inspection, source listing, etc.

Reply:
None of that functionality interacts with reverse execution, so I don't think we need to test it in combination with reverse execution in LLDB. For most purposes reverse execution behaves just like forward execution from LLDB's point of view: there's a stop, then LLDB reads the new target state (memory and registers, mainly) to display information to the user.

(There are certainly issues that aren't covered here, e.g. handling the creation and destruction of threads, handling the loading and unloading of shared libraries, and handling watchpoints all interact with reverse execution. I plan to address some of these issues in future work, with their own tests. I'm trying to keep this PR as minimal as possible.)